File: Samples.Objc/Samples.ObjC/API/Responses/AccessTokenResponse.m
//
//  AccessTokenResponse.m
//  PayPalCheckout-Samples-iOS-Objc
//
//  Created by Haider Khan on 5/18/20.
//  Copyright © 2020 PayPal. All rights reserved.
//

#import "AccessTokenResponse.h"

@implementation AccessTokenResponse

- (id)initWithData:(NSData *)data {
  self = [super init];
  if (self) {
    NSError *error;
    NSDictionary *dictionaryFromData = [NSJSONSerialization
                                        JSONObjectWithData:data
                                        options:NSJSONReadingAllowFragments
                                        error:&error];
    if (error == nil) {
      if ([dictionaryFromData valueForKey:@"access_token"]) {
        NSString *accessToken = [[dictionaryFromData valueForKey:@"access_token"] stringValue];
        self.accessToken = accessToken;
      }
    }
    else {
      self.accessToken = @"";
    }
  }
  return self;
}

@end

File: Samples.Swift/Samples.Swift/API/Responses/AccessTokenResponse.swift
//
//  AccessTokenResponse.swift
//  paypalcheckout-iOS-swift-sample
//
//  Created by Jax DesMarais-Leder on 3/16/21.
// Copyright © 2021 PayPal. All rights reserved.
//

import Foundation

struct AccessTokenResponse: Codable {
  let accessToken: String

  enum CodingKeys: String, CodingKey {
    case accessToken = "access_token"
  }
}
